Wrappers/Python/ccpi/optimisation/operators/BlockOperator.py

```diff
@@ -0,0 +1,744 @@
+            return type(self)(*[ el.multiply(ot, out, *args, **kwargs) for el,ot in zip(self.containers,other)])
+        elif isinstance(other, numpy.ndarray):           
+            return type(self)(*[ el.multiply(ot, out, *args, **kwargs) for el,ot in zip(self.containers,other)])
+        return type(self)(*[ el.multiply(ot, out, *args, **kwargs) for el,ot in zip(self.containers,other.containers)])
+    
+    def divide(self, other , out=None ,*args, **kwargs):
+        self.is_compatible(other)
+        if isinstance(other, Number):
+            return type(self)(*[ el.divide(other, out, *args, **kwargs) for el in self.containers])
+        elif isinstance(other, list) or isinstance(other, numpy.ndarray):
+            return type(self)(*[ el.divide(ot, out, *args, **kwargs) for el,ot in zip(self.containers,other)])
+        return type(self)(*[ el.divide(ot, out, *args, **kwargs) for el,ot in zip(self.containers,other.containers)])
+    
+    def power(self, other , out=None, *args, **kwargs):
+        assert self.is_compatible(other)
+        if isinstance(other, Number):
+            return type(self)(*[ el.power(other, out, *args, **kwargs) for el in self.containers])
+        elif isinstance(other, list) or isinstance(other, numpy.ndarray):
+            return type(self)(*[ el.power(ot, out, *args, **kwargs) for el,ot in zip(self.containers,other)])
+        return type(self)(*[ el.power(ot, out, *args, **kwargs) for el,ot in zip(self.containers,other.containers)])
+    
+    def maximum(self,other, out=None, *args, **kwargs):
+        assert self.is_compatible(other)
+        if isinstance(other, Number):
+            return type(self)(*[ el.maximum(other, out, *args, **kwargs) for el in self.containers])
+        elif isinstance(other, list) or isinstance(other, numpy.ndarray):
+            return type(self)(*[ el.maximum(ot, out, *args, **kwargs) for el,ot in zip(self.containers,other)])
+        return type(self)(*[ el.maximum(ot, out, *args, **kwargs) for el,ot in zip(self.containers,other.containers)])
+    
+    ## unary operations    
+    def abs(self, out=None, *args,  **kwargs):
+        return type(self)(*[ el.abs(out, *args, **kwargs) for el in self.containers]) 
+    def sign(self, out=None, *args,  **kwargs):
+        return type(self)(*[ el.sign(out, *args, **kwargs) for el in self.containers])
+    def sqrt(self, out=None, *args,  **kwargs):
+        return type(self)(*[ el.sqrt(out, *args, **kwargs) for el in self.containers])
+    def conjugate(self, out=None):
+        return type(self)(*[el.conjugate() for el in self.containers])
+    
+    ## reductions
+    def sum(self, out=None, *args, **kwargs):
+        return numpy.asarray([ el.sum(*args, **kwargs) for el in self.containers])
+    def squared_norm(self):
+        y = numpy.asarray([el.squared_norm() for el in self.containers])
+        return y.sum() 
+    def norm(self):
+        y = numpy.asarray([el.norm() for el in self.containers])
```

Comment:
I believe `norm` should instead return numpy.sqrt(self.squared_norm()).

Reply:
that's right